### DISCUSSION TITLE
Projects/Windows Directory removed in version 8.8.0 and 8.9.0

### DISCUSSION THREAD

**Discussion by sabyasachi07** [curl/curl#14522] [General]
Hi Team,

As I was using Visual Studio to generate the dll for my application, same like for earlier versions(<8.8.0). Any reason we remove this path Projects/Windows in latest packages? if Yes, any alternative to create these files.

**Comment by bagder** [comment]
cmake can generate project files for Visual Studio

**Reply by sabyasachi07** [reply]
Can you please  send the command if I use visual studio in my windows?

**Reply by bagder** [reply]
> Can you please  send the command if I use visual studio in my windows?

cmake -B Builds -G 'Visual Studio 17 2022'

According to [this](https://discourse.cmake.org/t/how-to-use-cmake-to-generate-visual-studio-sln-and-vcxproj/5878).

**Reply by sabyasachi07** [reply]
Done thanks 🙏

**Reply by sabyasachi07** [reply]
The solution file generating for x64 bit configuration can we generate it for x86 bit also?

**Reply by talregev** [reply]
from here:
https://stackoverflow.com/questions/31148943/option-to-force-either-32-bit-or-64-bit-build-with-cmake
> cmake . -DCMAKE_GENERATOR_PLATFORM=x64

I guess you try x86.
In the link there is other good answers.

**Reply by sabyasachi07** [reply]
I was using libcurl 8.6.0 binaries  in my application and there is no SQL error but with 8.9.1 dll the error is like named pipes provider : couldn't open a connection to SQL server [53]. Did I miss any flags in cmake or any configuration needed to be change for libcurl solution ?

**Reply by sabyasachi07** [reply]
I am using this  command to generate solution file as Dll Release - Dll OpenSSL mode but it's not creating as expected for VS anything else needs to be modified.

cmake -G "Visual Studio 16 2019" -A win32 -D BUILD SHARED_LIBS=ON -D CMAKE_USE_OPENSSL-ON -D CMAKE_BUILD_TYPE="DLL Release - DLL OpenSSL" -D OPENSSL_ROOT_DIR="C:\MediaFiles\dlo_sdk\OpenSSL" -D OPENSSL LIBRARIES="C:\MediaFiles\dlo_sdk\OpenSSL\x86\lib" -D OPENSSL_INCLUDE_DIR="C:\MediaFiles\dlo_sdk\OpenSSL\x86\include\openss I"-B curl-buildx86

**Reply by talregev** [reply]
> -DCMAKE_BUILD_TYPE="DLL Release - DLL OpenSSL"

this is not valid string. It should be "Release"
Why you don't add: 
"-DCMAKE_GENERATOR_PLATFORM=x86"
?

**Reply by sabyasachi07** [reply]
For 1st question 
Ok, I will change to "release" string but in 8.6.0 package it comes with DLL Release - DLL OpenSSL, soi thought using this could create DCMAKE_BUILD_TYPE="DLL Release - DLL OpenSSL".
For 2nd question https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2017%202022.html here it is mentioned as Win32 so I was using it

**Reply by talregev** [reply]
this should be good one:
cmake -G "Visual Studio 17 2022" -A Win32
So maybe your solution is better.

**Reply by talregev** [reply]
It case sensitive, it should be "Release"
https://stackoverflow.com/questions/48754619/what-are-cmake-build-type-debug-release-relwithdebinfo-and-minsizerel
I think for msvc it not have effect, only when you build, not generate.

**Reply by talregev** [reply]
My suggestion is to try with your approach and "Release" string
Then try stackoverflow approach: 
by remove the "-A Win32" and add 
"-DCMAKE_GENERATOR_PLATFORM=x86"

Let us know what you find.

**Reply by sabyasachi07** [reply]
Using -A option it can generate but using DCMAKE_GENERATOR_PLATFORM=x86 it shows error The platform doesn't contain the configuration and platform combination of Debug|win32

**Reply by talregev** [reply]
I understand. and It still not solve your problem?
I am using vcpkg to compile curl. It can also compile in x86. Not sure it fit to your case, but I can explain more if you want:
https://github.com/microsoft/vcpkg
It will compile curl and all dependencies in the triplet you will choose. default is x64. It can be change to x86.

**Reply by sabyasachi07** [reply]
No it didn't fix my issue but in warning it shows
Manually-specified variables were not used by the project.
CMAKE_USE_OPENSSL
OPENSSL_INCLUDE_DIR
OPENSSL_ROOT_DIR
OPENSSL LIBRARIES
Did libcurl remove these dependencies from newer packages or anything else?

**Reply by talregev** [reply]
It should be:
-DCURL_USE_OPENSSL=ON
https://github.com/curl/curl/blob/curl-8_9_0/CMakeLists.txt#L412C24-L412C41

Other maybe come from find_packge

**Reply by sabyasachi07** [reply]
Okay, just a curiosity can we generate the windows folder and the project fles inside it using the batch files inside projects folder of curl in 8.9.0 package.  in 8.6.0 package it already comes with package but in newer it is not. We use cmake but has some issues in it.

**Reply by jay** [reply]
The pregenerated project files are for old legacy versions of Visual Studio. For newer versions of Visual Studio you should be able to build them the same by using cmake to generate the Visual Studio project files.

For example, from a 64-bit Visual Studio command prompt you could build the 64-bit version:

cmake ..\..\curl -A x64 -DCURL_USE_SCHANNEL=ON -DCURL_ZLIB=OFF -DCMAKE_INSTALL_PREFIX:PATH=X:\j\curl\cmake\curl-root-x64
cmake --build . --target INSTALL --config Release

I'm sure you could do the same for the 32-bit version from a Visual Studio 32-bit command prompt by using -A Win32. If you have a commant prompt open with the build environment set to Visual Studio 64-bit and you try to build 32-bit libcurl you may see weird __imp errors IIRC

If you really must use the pregenerated project files then you would have to do it by cloning the repo and running generate.bat. It only generates old versions of Visual Studio projects., like VS 2010 2013 etc. I think that in retrospect maybe we should have left the old pregenerated files in the release because if the user has Visual Studio 2010 or whatever in some old Windows and can't use cmake then it's more work for them to clone the repo to generate the project files.

**Reply by sabyasachi07** [reply]
For Debug mode it will be --config Debug right?

**Reply by jay** [reply]
> For Debug mode it will be --config Debug right?

Yes

**Reply by sabyasachi07** [reply]
Can we make the generated build file as system independent one. In my case suppose i generated the build using cmake in C:\Users\Admin\New1, When i use the generated build in other system path like E:\New folder\Build  to build my application using Visual studio 2019 v16 it is still searching for  "C:\Users\Admin\New1" in visual studio.

 In my local system path "C:\Users\Admin\New1" i ran these two commands 
 1.cmake -A x64 -DCURL_USE_SCHANNEL=ON -DCURL_ZLIB=OFF -DCURL_USE_OPENSSL=ON -DOPENSSL_INCLUDE_DIR=" 
  C:/path/to/openssl/64bit/include" -DOPENSSL_LIBRARIES="C:/path/to/openssl/64bit/lib" -DOPENSSL_ROOT_DIR=" 
  C:/path/to/openssl/64bit" -B build_directory

  cmake --build curl_buildx64 --config Release
  
  After generating the build files in curl_buildx64 , I copied the curl_buildx64 into other system path "E:\New folder\Build". Then i used this build folder inside my application to build it using Visual studio 2019 v16 it is still referencing to my old path("C:\Users\Admin\New1"). Could you please guide me to find the generic solution.
  
 !!Thanks in advance

**Reply by jzakrzewski** [reply]
You shouldn't copy the build directory, but install the project into specific prefix (there are other options, but they require better understanding of what's going on).

So what you **probably** want is something like:
```
cmake -A x64 -DCURL_USE_SCHANNEL=ON -DCURL_ZLIB=OFF -DCURL_USE_OPENSSL=ON -DOPENSSL_INCLUDE_DIR="
C:/path/to/openssl/64bit/include" -DOPENSSL_LIBRARIES="C:/path/to/openssl/64bit/lib" -DOPENSSL_ROOT_DIR="
C:/path/to/openssl/64bit" -B curl_buildx64 -DCMAKE_INSTALL_PREFIX="E:/New folder/Build"
cmake --build curl_buildx64 --config Release --target install
```
